api/api/test/test_authentication.py

```diff
@@ -126,15 +128,15 @@ def test_get_security_conf():
     assert all(x in result.keys() for x in ('auth_token_exp_timeout', 'rbac_mode'))
 
 
+@pytest.mark.asyncio
 @pytest.mark.parametrize('auth_context', [{'name': 'initial_auth'}, None])
 @patch('api.authentication.jwt.encode', return_value='test_token')
 @patch('api.authentication.generate_keypair', return_value=('-----BEGIN PRIVATE KEY-----',
                                                             '-----BEGIN PUBLIC KEY-----'))
 @patch('wazuh.core.cluster.dapi.dapi.DistributedAPI.__init__', return_value=None)
 @patch('wazuh.core.cluster.dapi.dapi.DistributedAPI.distribute_function', side_effect=None)
-@patch('concurrent.futures.ThreadPoolExecutor.submit', side_effect=None)
 @patch('api.authentication.raise_if_exc', side_effect=None)
-def test_generate_token(mock_raise_if_exc, mock_submit, mock_distribute_function, mock_dapi, mock_generate_keypair,
+async def test_generate_token(mock_raise_if_exc, mock_distribute_function: AsyncMock, mock_dapi, mock_generate_keypair,
```

Comment:
If you are going to add typings, please add for all the parameters.

Reply:
Since all the other tests' mocks do not have type hints, we will remove them